--- pixman-0.18.0/pixman/pixman-compiler.h.orig	2010-03-28 00:39:02.000000000 +0100
+++ pixman-0.18.0/pixman/pixman-compiler.h	2010-04-21 23:26:50.704991410 +0200
@@ -70,7 +70,14 @@
 #endif
 
 /* TLS */
-#if defined(TOOLCHAIN_SUPPORTS__THREAD)
+#if defined(PIXMAN_NO_TLS)
+
+#   define PIXMAN_DEFINE_THREAD_LOCAL(type, name)			\
+    static type name
+#   define PIXMAN_GET_THREAD_LOCAL(name)				\
+    (&name)
+
+#elif defined(TOOLCHAIN_SUPPORTS__THREAD)
 
 #   define PIXMAN_DEFINE_THREAD_LOCAL(type, name)			\
     static __thread type name
